psvgxp/SceGxmTypes.h

```diff
@@ -104,22 +104,20 @@ typedef enum SceGxmParameterType
 
 typedef struct SceGxmProgramParameter // SceGxmProgramParameter
 {
-   std::uint32_t name_offset; //offset from here to string name
-
-   std::uint32_t type;   //[unused(16):container_index(4):component_count(4):parameter_type(4):category(4)]
-                         //container_index - applicable for constants, not applicable for samplers (buffer, default, texture)
-                         //component_count -  applicable for constants, not applicable for samplers (select size like float2, float3, float3 ...)
-                         //parameter_type -  applicable for constants, not applicable for samplers (select type like float, half, fixed ...)  - SceGxmParameterType
-                         //category -  select constant or sampler - SceGxmParameterCategory
-
-   std::uint32_t array_size; //number of elements in array (applicable for constants, not applicable for samplers) - this confirmed to be unsigned integer
-   std::int32_t resource_index; //register number (applicable for constants, applicable for samplers) - this confirmed to be signed integer
-}SceGxmProgramParameter;
+   std::uint32_t name_offset; // offset from here to string name
+   std::uint16_t category : 4; // SceGxmParameterCategoryselect constant or sampler
+   std::uint16_t type : 4; // SceGxmParameterType - applicable for constants, not applicable for samplers (select type like float, half, fixed ...)
+   std::uint16_t component_count : 4; // applicable for constants, not applicable for samplers (select size like float2, float3, float3 ...)
+   std::uint16_t container_index : 4; // applicable for constants, not applicable for samplers (buffer, default, texture)
+   std::uint16_t unknown : 4;
```

Comment:
should not this be just  std::uint16_t  ? Originally it was unused(16) and type was a 32 bit field.